src/DotNetWorker.Grpc/FunctionMetadata/FunctionMetadataJsonReader.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+
+namespace Microsoft.Azure.Functions.Worker
+{ 
+    internal class FunctionMetadataJsonReader
+    {
+        private readonly string _directory;
+        private const string FileName = "functions.metadata";
+
+        public FunctionMetadataJsonReader(string directory)
+        {
+            _directory = directory ?? throw new ArgumentNullException(nameof(directory));
+        }
+
+        public virtual async Task<ImmutableArray<FunctionMetadata>> ReadMetadataAsync()
+        {
+            string metadataFile = Path.Combine(_directory, FileName);
+
+            if (File.Exists(metadataFile))
+            {
+                using (var fs = File.OpenText(metadataFile))
+                {
+                    using (var js = new JsonTextReader(fs))
+                    {
+                        JArray functionMetadataJson = (JArray)await JToken.ReadFromAsync(js);
```

Comment:
I think @brettsam mentioned this above, but it would be good to avoid a dependency on JSON.NET here.